File: f77/driver/limit.f
C
      Real Function Limit(X, Lower_Limit, Upper_Limit)
C
C     This function bounds X between Lower_Limit and Upper_Limit.
C
C     Inputs:
C       Lower_Limit - Minimum allowable value for Value [Real]
C       Upper_Limit - Maximum allowable value for Value [Real]
C       X           - The value to be limited [Real]
C
C     Outputs:
C       Limit  -  Bounded value of X [Real]
C
C     Internal variables and constants:
C       None.
C
      Implicit None
      Real Lower_Limit,
     &     Upper_Limit,
     &     X
C
      If (X .lt. Lower_Limit) Then
        Limit = Lower_Limit
      Else If (X .gt. Upper_Limit) Then
        Limit = Upper_Limit
      Else
        Limit = x
      End If
C
      Return
      End

File: compcert/driver/limit.c
#include "limit.h"
REAL limit( REAL x           // X           - The value to be limited [Real]
          , REAL lower_limit // Lower_Limit - Minimum allowable value for Value [Real]
          , REAL upper_limit // Upper_Limit - Maximum allowable value for Value [Real]
          )
{
  // Outputs:
  //    Limit  -  Bounded value of X [Real]
  if      (x < lower_limit) return lower_limit;
  else if (x > upper_limit) return upper_limit;
  else return x;
}
